### DISCUSSION TITLE
Automatic1111 Backend and overall impressions

### DISCUSSION THREAD

**Discussion by rlewisfr** [Stability-AI/StableSwarmUI#220] [General]
Not much activity happening on this discussion platform, although McMonkey seems to be active (thumbs up).

I think I managed to get the A1111 backend up and running, properly configured.  I pointed the server start up script to my A1111 install, and used my parameters `-allow-code --xformers --no-half-vae --medvram-sdxl --ckpt-dir "I:\Stable Diffusion Model Library\Stable-diffusion"`.  Not sure if I needed to specify the ckpt dir, but I get some strange results from Swarm related to Model directories at times so thought I would include.

Question is:  is the A1111 integration still very basic at this Alpha point or am I missing something?  If I would rate the ComfyUI integration at an "Intermediate" level, A1111 appears to be "Basic/Novice".  Swarm appears to be able to trigger the backend with the appropriate parameters (i.e. Swarm-->A1111) but nothing in reverse (i.e. A1111-->Swarm).  For instance, no options for Adetailer, Reactor, etc. etc.

**I have to say that I'm initially very impressed with Swarm.**  It has some really cool functions that I like so far:

- Love the grid compare options
- Presets are a god send!
- Being able to "open the hood" (for ComfyUI) to tweak and change is nice, although I'm not really a ComfyUI guy
- Like being able to hide and resize panels easily

Keep up the good work!

**Comment by mcmonkey4eva** [comment]
The auto webui backend support hasn't received much time/attention as, frankly, it doesn't really offer any advantage to use auto as a backend over comfy (the only theoretical advantage would be the decently large pre-existing extension ecosystem... which has no standard usage format and so couldn't really be automatically forwarded to swarm),
meanwhile comfy has a massive pile of advantages (eg better performance, better automation of installation, better configurability, much wider range of customization, much more room for addons to integrate well, much more reliable and clear API, the ability to do cool things like custom workflows, ...)

**Reply by CodeAlexx** [reply]
been using the front end for a time on and off, i miss ip-adapter, animatediff style as Auto1111. Unless It is there and I am missing something or don't know how to get it. you have an incredible product!!

**Reply by mcmonkey4eva** [reply]
IP-Adapter is a little bit weird atm, not fully integrated but it is supported - if you install the comfy IP-Adapter addon https://github.com/cubiq/ComfyUI_IPAdapter_plus and download ip-adapter models and restart swarm, you can then drag images into the prompt box to use them with ip-adapter.

AnimateDiff doesn't have full integration atm, but you can at least use a comfy addon for it ( https://github.com/Kosinkadink/ComfyUI-AnimateDiff-Evolved or https://github.com/ArtVentureX/comfyui-animatediff ) and grab a workflow and import that workflow to swarm.

In the future I'd like AnimateDiff to be supported as an option in the Video controls, but for now that's SVD-based video gen only. I'd also like IP-Adapter to be out-of-the-box supported but atm you gotta download it separately.
